### PR TITLE
chore(rpc): generate latest spec

### PR DESCRIPTION
**Pull request by LucasLvy**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Bugfix
- Feature
- Code style update (formatting, renaming)
- Refactoring (no functional changes, no API changes)
- Build-related changes
- Documentation content changes
- Testing
- Other (please describe):

## What is the current behavior?

Resolves: #NA

## What is the new behavior?

-
-
-

### CODE REVIEW

**Review by LucasLvy — COMMENTED** [review]
I think we should require the felt type to impl Debug, Display, Clone, Serialize, Deserialize

File: crates/starknet-types-rpc/src/v0_7_1/starknet_api_openrpc.rs [hunk 1]
```diff
@@ -0,0 +1,2886 @@
...
+    /// specifies whether the data of this block is published via blob data or calldata
+    pub l1_da_mode: L1DaMode,
+    /// The price of l1 data gas in the block
+    pub l1_data_gas_price: ResourcePrice<F>,
+    /// The price of l1 gas in the block
+    pub l1_gas_price: ResourcePrice<F>,
+    /// The hash of this block's parent
+    pub parent_hash: BlockHash<F>,
+    /// The StarkNet identity of the sequencer submitting this block
+    pub sequencer_address: F,
```
Comment (tdelabro) [inline crates/starknet-types-rpc/src/v0_7_1/starknet_api_openrpc.rs:741]: Address

File: crates/starknet-types-rpc/src/v0_7_1/starknet_write_api.rs [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct ClassAndTxnHash<F> {
+    pub class_hash: F,
+    pub transaction_hash: TxnHash<F>,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct ContractAndTxnHash<F> {
+    pub contract_address: Felt,
```
Comment (tdelabro) [inline crates/starknet-types-rpc/src/v0_7_1/starknet_write_api.rs:19]: F or Address

**Review by tdelabro — CHANGES_REQUESTED** [review]

File: crates/starknet-types-rpc/src/custom/query.rs [hunk 3]
```diff
@@ -7,42 +7,42 @@ use crate::{
     BroadcastedDeclareTxnV1, BroadcastedDeclareTxnV2, DeployAccountTxnV1, InvokeTxnV0, InvokeTxnV1,
 };
 
-#[derive(Debug, Clone, Serialize, Deserialize)]
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
 #[serde(tag = "version")]
-pub enum BroadcastedDeclareTxn {
+pub enum BroadcastedDeclareTxn<F: Default> {
```
[the remark this answers is not in the document]
Comment (maciejka) [reply crates/starknet-types-rpc/src/custom/query.rs:13]: Added V3 related declarations.

File: crates/starknet-types-rpc/src/v0_7_1/starknet_api_openrpc.rs [hunk 4]
```diff
@@ -0,0 +1,2886 @@
+//
+// This file was automatically generated by openrpc-gen first, then manually edited.
```
[the remark this answers is not in the document]
Comment (maciejka) [reply crates/starknet-types-rpc/src/v0_7_1/starknet_api_openrpc.rs]: Now it is fully automatic.

File: crates/starknet-types-rpc/src/v0_7_1/starknet_api_openrpc.rs [hunk 5]
```diff
@@ -0,0 +1,2886 @@
...
+    #[serde(rename = "EXTERNAL")]
+    pub external: Vec<SierraEntryPoint<F>>,
+    #[serde(rename = "L1_HANDLER")]
+    pub l1_handler: Vec<SierraEntryPoint<F>>,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct ContractStorageDiffItem<F: Default> {
+    /// The contract address for which the storage changed
+    pub address: F,
```
[the remark this answers is not in the document]
Comment (maciejka) [reply crates/starknet-types-rpc/src/v0_7_1/starknet_api_openrpc.rs:300]: It is declared as felt in the spec. I added a pr to the spec: https://github.com/starkware-libs/starknet-specs/pull/213.

**Review by maciejka — APPROVED** [review]

**Review by tdelabro — APPROVED** [review]

**Pull request merged**